internal/component/pyroscope/relabel/metrics.go

```diff
@@ -12,41 +17,41 @@ type metrics struct {
 }
 
 func newMetrics(reg prometheus.Registerer) *metrics {
```

Comment:
Not necessarily blocking, but it is super easy to mess up the metrics, so I would suggest you strongly consider adding some test coverage for it, it is fairly straight forward: https://pkg.go.dev/github.com/prometheus/client_golang/prometheus/testutil#CollectAndCompare

Reply:
Updated. I also took the chance to add a test checking the purge of the cache.